api/src/main/java/com/github/retrooper/packetevents/protocol/packettype/PacketType.java

```diff
@@ -505,35 +558,35 @@ public PacketSide getSide() {
 
         public enum Server implements PacketTypeCommon, ClientBoundPacket {
 
-            PLUGIN_MESSAGE,
-            DISCONNECT,
-            CONFIGURATION_END,
-            KEEP_ALIVE,
-            PING,
-            REGISTRY_DATA,
-            RESOURCE_PACK_SEND,
-            UPDATE_ENABLED_FEATURES,
-            UPDATE_TAGS,
+            PLUGIN_MESSAGE(WrapperConfigServerPluginMessage.class),
+            DISCONNECT(WrapperConfigServerDisconnect.class),
+            CONFIGURATION_END(WrapperConfigServerConfigurationEnd.class),
+            KEEP_ALIVE(WrapperConfigServerKeepAlive.class),
+            PING(null),
+            REGISTRY_DATA(WrapperConfigServerRegistryData.class),
+            RESOURCE_PACK_SEND(WrapperConfigServerResourcePackSend.class),
+            UPDATE_ENABLED_FEATURES(WrapperConfigServerUpdateEnabledFeatures.class),
+            UPDATE_TAGS(null),
 
             // Added in 1.20.3
-            RESOURCE_PACK_REMOVE,
+            RESOURCE_PACK_REMOVE(WrapperConfigServerResourcePackRemove.class),
 
             // Added in 1.20.5
-            COOKIE_REQUEST,
-            RESET_CHAT,
-            STORE_COOKIE,
-            TRANSFER,
-            SELECT_KNOWN_PACKS,
+            COOKIE_REQUEST(WrapperConfigServerCookieRequest.class),
+            RESET_CHAT(WrapperConfigServerResetChat.class),
+            STORE_COOKIE(WrapperConfigServerStoreCookie.class),
+            TRANSFER(WrapperConfigServerTransfer.class),
+            SELECT_KNOWN_PACKS(WrapperConfigServerSelectKnownPacks.class),
 
             // added in 1.21
-            CUSTOM_REPORT_DETAILS,
-            SERVER_LINKS;
+            CUSTOM_REPORT_DETAILS(WrapperConfigServerCustomReportDetails.class),
+            SERVER_LINKS(WrapperConfigServerServerLinks.class);
 
             private static int INDEX = 0;
             private static final Map<Byte, Map<Integer, PacketTypeCommon>> PACKET_TYPE_ID_MAP = new HashMap<>();
             private final int[] ids;
 
-            Server() {
+            Server(@Nullable Class<?> clazz) {
```

Comment:
```suggestion
            Server(@Nullable Class<? extends PacketWrapper<?>> clazz) {
```

Also, this doesn't seem to be saved + exposed as a field